### DISCUSSION TITLE
Google Pixel 3 XL (crosshatch) stuck at Google Logo after flash

### DISCUSSION THREAD

**Discussion by eNTi** [Divested-Mobile/DivestOS-Build#128] [Q&A]
Hi,

I've tried to follow the installation guide [here](https://divestos.org/index.php?page=bootloader) although I'm confused with step 8. The package is there but the site shows that the firmware is enabled... so should I run that or not? Regardless if I should I tried both and it's the same every time: After 12. the device is just stuck at a white screen showing the google logo.

Any help is appreciated.

**Comment by SkewedZeppelin** [comment]
> Status: Broken

Pixel 3 and up are not booting along with OnePlus 9 and Poco F3
I don't know why and I don't have any to debug it.

**Reply by eNTi** [reply]
Can I help you somehow?

**Reply by martin-braun** [reply]
The Pixel 3a is out of support now and the only highly maintained compatible long-life ROM I found is LineageOS and [it's security is terrible](https://madaidans-insecurities.github.io/android.html#lineageos). DivestOS is the best privacy focused fork even mentioned by The Hated One as recommend fallback, if GrapheneOS cannot be installed. 

Managing to support Pixel phones that are out of support will boost attention on DivestOS by Pixel users who prefer longevity. My Pixel 3a is fine as a daily driver, it serves my need, yet I should replace it? Although DivestOS cannot offer firmware updates, it can still provide security patches longer than GrapheneOS will. 

Running a privacy focused Android 13 with a locked bootloader on a Pixel 3a would be a huge selling point for DivestOS. I wonder how LineageOS passes, but DivestOS does not. I hope there will be a way to solve this issue anytime soon.

**Reply by SkewedZeppelin** [reply]
I've found it very hard to find affordably priced used and also bootloader unlockable Pixel's.
I've bought many over the years and all have been Verizon locked variants.

I wouldn't be surprised if whatever boot issue there is could be bisected in just a few hours, but without a device in my hand or someone volunteering to do the work it will remain unchanged.

**Reply by martin-braun** [reply]
@SkewedZeppelin I got a Pixel 3a without SIM lock and its OEM status can be unlocked as well. Although, it's my primary device at the moment, I would be willing to wipe it to test builds and give relevant info to get this going. I just need clear instructions. Is there a way to do it this way or does the person who fixes those issues need it in their hand?

**Reply by SkewedZeppelin** [reply]
So since it never gets past the Google logo it means the kernel is the likely issue. But it is also possible to be a signing issue, although unlikely.

To debug I'd first compile the kernel without any changes to it as a sanity check.
Then I'd apply the defconfig hardening.
Then move apply a handful of the kernel [patches](https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/LineageOS-19.1/CVE_Patchers/android_kernel_google_msm-4.9.sh) at a time.
And stop when it breaks.
Given that Pixel 4a is also reported to not boot, I suspect it may be defconfig related.

To have someone remotely try this I have to compile like ten kernels, note them down appropriately, and let someone flash them to narrow it down. Then repeat with some more kernels to narrow that down until it is just the single patch or option.

For the user they would just have to flash the latest DivestOS, and then `fastboot flash boot boot.img` for each test kernel.

**Reply by martin-braun** [reply]
Please confirm if I got it right: 

- The user flashes the latest DivestOS on their Pixel 3a => The device won't boot
- Then the user would flash only the kernel over that LineageOS uses => The device will boot
- Now the user keeps flashing each and every upcoming kernel you provide in which changes were added step by step to narrow down the point of failure

I definitely cannot do it now, but I can probably do it sometimes next week(end). How would we communicate and share files, etc.?

**Reply by martin-braun** [reply]
@SkewedZeppelin Can I get in touch with you via instant messaging? If a Sunday is suitable for you, we could try to get things working tomorrow.

**Reply by SkewedZeppelin** [reply]
@martin-braun 

Chat is documented here: https://divestos.org/index.php?page=community

However it is likely a solution to this has been found: https://github.com/Divested-Mobile/DivestOS-Build/commit/c97f8a1b6a6fc2103fa014c32a4d4f066f77b30e
https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/LineageOS-19.1/Patch.sh#L461

**Comment by ncorder** [comment]
@SkewedZeppelin If you are still looking for a device to test on, I own a Pixel 3a XL and am more than willing to help.

**Comment by SkewedZeppelin** [comment]
https://github.com/Divested-Mobile/DivestOS-Build/commit/06254708be4afc24c3158c8c145adca35a58ccee likely fixes this

**Reply by eNTi** [reply]
Sadly it doesn't look like it.

**Reply by SkewedZeppelin** [reply]
Does it reboot to fastbootd (green screen) successfully and finish the fastboot update?

**Reply by eNTi** [reply]
From 9. on it will install the package then reboot directly into the logo where it will hang. I've tried several avenues to get the other package installed but the only thing that ever really works it the adb installation from the lineageos recovery. I've restored stock android but all I get from THAT recovery is a signature failure.

Even then... with the lineageos recovery it will install but then hang at the logo again.

**Reply by SkewedZeppelin** [reply]
Thank you for testing again @eNTi
